File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere_onnx/android/src/main/kotlin/ai/runanywhere/sdk/onnx/OnnxPlugin.kt
package ai.runanywhere.sdk.onnx

import android.os.Build
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/**
 * RunAnywhere ONNX Flutter Plugin - Android Implementation
 *
 * This plugin provides the native bridge for the ONNX backend on Android.
 * The actual STT/TTS/VAD functionality is provided by RABackendONNX native libraries (.so files).
 */
class OnnxPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    companion object {
        private const val CHANNEL_NAME = "runanywhere_onnx"
        private const val BACKEND_VERSION = "0.1.4"
        private const val BACKEND_NAME = "ONNX"

        init {
            // Load ONNX backend native libraries
            try {
                System.loadLibrary("onnxruntime")
                System.loadLibrary("sherpa-onnx-c-api")
                System.loadLibrary("rac_backend_onnx_jni")
            } catch (e: UnsatisfiedLinkError) {
                // Library may not be available in all configurations
                android.util.Log.w("ONNX", "Failed to load ONNX libraries: ${e.message}")
            }
        }
    }

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, CHANNEL_NAME)
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "getPlatformVersion" -> {
                result.success("Android ${Build.VERSION.RELEASE}")
            }
            "getBackendVersion" -> {
                result.success(BACKEND_VERSION)
            }
            "getBackendName" -> {
                result.success(BACKEND_NAME)
            }
            "getCapabilities" -> {
                result.success(listOf("stt", "tts", "vad"))
            }
            else -> {
                result.notImplemented()
            }
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere_onnx/ios/Classes/OnnxPlugin.swift
import Flutter
import UIKit

/// RunAnywhere ONNX Flutter Plugin - iOS Implementation
///
/// This plugin provides the native bridge for the ONNX backend on iOS.
/// The actual STT/TTS/VAD functionality is provided by RABackendONNX.xcframework.
public class OnnxPlugin: NSObject, FlutterPlugin {

    public static func register(with registrar: FlutterPluginRegistrar) {
        let channel = FlutterMethodChannel(
            name: "runanywhere_onnx",
            binaryMessenger: registrar.messenger()
        )
        let instance = OnnxPlugin()
        registrar.addMethodCallDelegate(instance, channel: channel)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
        case "getPlatformVersion":
            result("iOS " + UIDevice.current.systemVersion)
        case "getBackendVersion":
            result("0.1.4")
        case "getBackendName":
            result("ONNX")
        case "getCapabilities":
            result(["stt", "tts", "vad"])
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}
